units/systemd-timesyncd.service.in

```diff
@@ -22,6 +22,10 @@ Wants=time-set.target
 AmbientCapabilities=CAP_SYS_TIME
 BusName=org.freedesktop.timesync1
 CapabilityBoundingSet=CAP_SYS_TIME
+# Turn off DNSSEC validation for hostname look-ups, since those need the
+# correct time to work, but we likely won't acquire that without NTP. Let's
+# break this chicken-and-egg cycle here.
+Environment=SYSTEMD_NSS_RESOLVE_VALIDATE=0
```

Comment:
Why not set the variable dynamically? That is, how about to retry with the flag when an hostname look-up failed due to DNSSEC?

Reply:
but that would not be any safer than not doing dnssec at all for this. it would man we'd try dnssec nd then ignore the results... (which is different from opportunistic dnssec where we try dnssec and only if we have reason to believe the servers aren't capable of it ignore it).

what might make sense is to add another mode that does full dnssec vlidation except for one thing: the time-based validation of signatures. But that's harder than it might look at first, because we'd need another set of caches for this, since we can't mix stuff validated this way with neither stuff that was validated fully, not the stuff that was not validated at all. 

I am not sure the effort for that would be worth it. (and not caching for this form of half-baked dnssec wouldn't be an option either, since dnssec is super slow without caching)

I think the right approach to secure ntp would be implementing ntpsec I guess, i.e. actually protect the clock sync logic, and not the way to figure out the server. if we do ntpsec properly, then the dnssec validation of the hostname doesn't really matter.

Reply:
> but that would not be any safer than not doing dnssec at all for this. it would man we'd try dnssec nd then ignore the results... (which is different from opportunistic dnssec where we try dnssec and only if we have reason to believe the servers aren't capable of it ignore it).

Yeah, that makes sense to me. Thank you for the explanation.